lib/base/context.hpp

```diff
@@ -33,18 +34,21 @@ std::ostream& operator<<(std::ostream& stream, const ContextTrace& trace);
 class ContextFrame
 {
 public:
-	ContextFrame(const String& message);
+	ContextFrame(std::function<void(std::ostream&)> message);
 	~ContextFrame();
 
 private:
-	static std::list<String>& GetFrames();
+	static std::vector<std::function<void(std::ostream&)>>& GetFrames();
 
 	friend class ContextTrace;
 };
 
 /* The currentContextFrame variable has to be volatile in order to prevent
  * the compiler from optimizing it away. */
-#define CONTEXT(message) volatile icinga::ContextFrame currentContextFrame(message)
+#define CONTEXT(message) volatile icinga::ContextFrame currentContextFrame ([&](std::ostream& db026162_4822_4144_95e0_4a5a42d1c1b3) { \
+db026162_4822_4144_95e0_4a5a42d1c1b3 << message; \
```

Comment:
That name is really not helpful at all.

Either go with something that is unlikely to conflict but still makes sense as a name (something like `_CONTEXT_stream`) or, if you want to remove all potential conflicts, go with the other version I suggested:

```c++
#define CONTEXT(message) volatile icinga::ContextFrame currentContextFrame ([&]() { \
    return (::std::stringstream() << message).str(); })
```